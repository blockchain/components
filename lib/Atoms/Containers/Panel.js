"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  display: flex;\n  flex-direction: column;\n  justify-content: center;\n  align-items: center;\n  height: ", ";\n  width: ", ";\n  box-shadow: 0 0 25px 5px ", ";\n  background-color: ", ";\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var BasePanel = _styledComponents.default.div(_templateObject(), function (props) {
  return props.height;
}, function (props) {
  return props.width;
}, function (props) {
  return props.theme['gray1'];
}, function (props) {
  return props.theme['white'];
});

var Panel = function Panel(_ref) {
  var children = _ref.children,
      height = _ref.height,
      width = _ref.width,
      rest = _objectWithoutProperties(_ref, ["children", "height", "width"]);

  return _react.default.createElement(BasePanel, _extends({
    height: height,
    width: width
  }, rest), children);
};

Panel.propTypes = {
  children: _propTypes.default.node,
  height: _propTypes.default.string,
  width: _propTypes.default.string
};
Panel.defaultProps = {
  height: 'auto',
  width: '100%'
};
var _default = Panel;
exports.default = _default;