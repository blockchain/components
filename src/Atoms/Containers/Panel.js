// @flow strict
import PropTypes from 'prop-types'
import * as React from 'react'
import styled from 'styled-components'

type PropsType = {
  +children?: React.Node,
  +height?: string,
  +width?: string,
}

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: ${(props) => props.height};
  width: ${(props) => props.width};
  min-width: ${(props) => props.width};
  box-shadow: 0 0 1.5rem 0.4rem ${(props) => props.theme['panelShadowColor']};
  background-color: ${(props) => props.theme['panelBackgroundColor']};
`

const Panel = ({ children, ...rest }: PropsType) => (
  <Wrapper {...rest}>{children}</Wrapper>
)

Panel.propTypes = {
  children: PropTypes.node,
  height: PropTypes.string,
  width: PropTypes.string,
}

Panel.defaultProps = {
  height: 'auto',
  width: '100%',
}

export default Panel
