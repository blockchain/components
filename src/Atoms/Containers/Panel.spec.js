import React from 'react'
import { mount } from 'enzyme'

import Panel from './Panel'

describe('Panel', () => {
  it('should render correctly', () => {
    const tree = mount(<Panel>MyPanel</Panel>)

    expect(tree).toMatchSnapshot()
  })

  it('should render correctly with width', () => {
    const tree = mount(<Panel width="200px">MyPanel</Panel>)

    expect(tree).toMatchSnapshot()
  })

  it('should render correctly with height', () => {
    const tree = mount(<Panel height="200px">MyPanel</Panel>)

    expect(tree).toMatchSnapshot()
  })
})
