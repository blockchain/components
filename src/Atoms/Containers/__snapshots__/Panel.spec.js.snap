// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: auto;
  width: 100%;
  min-width: 100%;
  box-shadow: 0 0 1.5rem 0.4rem #EEF2F5;
  background-color: #FFF;
}

<Panel
  height="auto"
  width="100%"
>
  <styled.div
    height="auto"
    width="100%"
  >
    <div
      className="c0"
      height="auto"
      width="100%"
    >
      MyPanel
    </div>
  </styled.div>
</Panel>
`;

exports[`Panel should render correctly with height 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 200px;
  width: 100%;
  min-width: 100%;
  box-shadow: 0 0 1.5rem 0.4rem #EEF2F5;
  background-color: #FFF;
}

<Panel
  height="200px"
  width="100%"
>
  <styled.div
    height="200px"
    width="100%"
  >
    <div
      className="c0"
      height="200px"
      width="100%"
    >
      MyPanel
    </div>
  </styled.div>
</Panel>
`;

exports[`Panel should render correctly with width 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: auto;
  width: 200px;
  min-width: 200px;
  box-shadow: 0 0 1.5rem 0.4rem #EEF2F5;
  background-color: #FFF;
}

<Panel
  height="auto"
  width="200px"
>
  <styled.div
    height="auto"
    width="200px"
  >
    <div
      className="c0"
      height="auto"
      width="200px"
    >
      MyPanel
    </div>
  </styled.div>
</Panel>
`;
